src/Integration/MefServices/MefSonarQubeService.cs

```diff
@@ -40,10 +41,34 @@ public class MefSonarQubeService : SonarQubeService
     {
         [ImportingConstructor]
         public MefSonarQubeService(ILogger logger)
-            : base(new LoggingHttpClientHandler(logger),
-                DefaultConfiguration.Configure(new RequestFactory(s => logger.LogDebug(s))),
-                userAgent: $"SonarLint Visual Studio/{VersionHelper.SonarLintVersion}")
+            : base(new HttpClientHandler(),
+                requestFactory: DefaultConfiguration.Configure(new RequestFactory(new LoggerAdapter(logger))),
+                userAgent: $"SonarLint Visual Studio/{VersionHelper.SonarLintVersion}",
+                logger: new LoggerAdapter(logger))
         {
         }
+
+        private class LoggerAdapter : SonarQube.Client.Helpers.ILogger
+        {
+            private readonly ILogger logger;
+
+            public LoggerAdapter(ILogger logger)
+            {
+                this.logger = logger;
+            }
+
+            public void Debug(string message) =>
+                // This will be executed only on DEBUG build.
+                logger.LogDebug(message);
+
+            public void Error(string message) =>
```

Comment:
It might be worth prefixing the _Error_ and _Warning_ messages with "Error: " and "Warning: " prefixes to give some indication about which messages deserve more notice.